### PR TITLE
feat: fire up test chains by first block: bitcoin#22818, dip1, dip8, dip20, brr - 3/n

### PR DESCRIPTION
**Pull request by knst**
## What was done?
Backport bitcoin#22818 which helped to activate all forks from block-1 at regtest.
Activate next dash's softforks at block 1:
 - DIP-0001 (blocksize 2mb)
 - DIP-0020 (opcodes)
 - DIP-0008 (chainlocks)
 - BRR (block reward reallocation)

## Breaking Changes
All changes are relevant to RegTest only

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see fde8ed17777eec038b41b13633bd804a94cb6360 and 55aa14e7bfd69e3d15137ec0c912e38dad0c930a

**Comment by knst** [comment]
@UdjinM6 , I would disagree with both patches.

https://github.com/dashpay/dash/commit/55aa14e7bfd69e3d15137ec0c912e38dad0c930a:


There are both values `bad-txns-oversize` and `tx-size` used in our (and bitcoin's) implementation.

`bad-txns-oversize`: https://github.com/dashpay/dash/blob/develop/src/consensus/tx_check.cpp#L34
`tx-size`: https://github.com/dashpay/dash/blob/develop/src/policy/policy.cpp#L73


I'd say inside `ContextualCheckTransaction` just has been used a wrong id for failure - let's correct it better rather than wrap it

----------------

https://github.com/dashpay/dash/commit/fde8ed17777eec038b41b13633bd804a94cb6360

That remove existing command line argument, I'd like to keep it for v21.2 just in case if anyone uses it.
I am going to create a new PR with removing bip147, dip3param, dip8param, etc when v22 will be branched-out.

**Comment by UdjinM6** [comment]
> @UdjinM6 , I would disagree with both patches.
> 
> [55aa14e](https://github.com/dashpay/dash/commit/55aa14e7bfd69e3d15137ec0c912e38dad0c930a):
> 
> There are both values `bad-txns-oversize` and `tx-size` used in our (and bitcoin's) implementation.
> 
> `bad-txns-oversize`: https://github.com/dashpay/dash/blob/develop/src/consensus/tx_check.cpp#L34 `tx-size`: https://github.com/dashpay/dash/blob/develop/src/policy/policy.cpp#L73
> 
> I'd say inside `ContextualCheckTransaction` just has been used a wrong id for failure - let's correct it better rather than wrap it in numerous of exceptions to make it disappear in just particular 2 tests

`bad-txns-oversize` is a consensus level error and `tx-size` is a mempool policy error, the one you are changing is consensus level one, it was picked intentionally that way.

> [fde8ed1](https://github.com/dashpay/dash/commit/fde8ed17777eec038b41b13633bd804a94cb6360)
> 
> That remove existing command line argument, I'd like to keep it for v21.2 just in case if anyone uses it. I am going to create a new PR with removing bip147, dip3param, dip8param, etc when v22 will be branched-out.

Sounds good

**Comment by knst** [comment]
> bad-txns-oversize is a consensus level error and tx-size is a mempool policy error, the one you are changing is consensus level one, it was picked intentionally that way.

Indeed, it makes sense.

Please check my force-push: I applied your changes and also dropped a global flag `fDIP0001ActiveAtTip` which is no need anymore

**Review by UdjinM6 — APPROVED** [review]
utACK

side note: thinking more about `-bip147height` and `-dip8params`.. they are regtest-only and replacing them with corresponding `-testactivationheight` cmd-line options is trivial. Dropping them would also prevent confusion when new and old options are used together accidentally. But I won't insist :)

File: test/functional/feature_dip3_v19.py [hunk 1]
```diff
@@ -57,8 +57,6 @@ def run_test(self):
             if i != 0:
                 self.connect_nodes(i, 0)
 
-        self.activate_dip8()
```
Comment (PastaPastaPasta) [inline test/functional/feature_dip3_v19.py:60]: why'd we drop this here?

Comment (PastaPastaPasta) [reply test/functional/feature_dip3_v19.py:60]: nvm; it activates at 1 unless specified otherwise

**Review by PastaPastaPasta — APPROVED** [review]
utACK f4ba2bb76906061469f33fad6f3b9ef39df4c5a6

**Pull request merged**